tests/unit_tests/data_steward/cdr_cleaner/cleaning_rules/null_invalid_foreign_keys.py

```diff
@@ -0,0 +1,139 @@
+"""
+Unit test for cleaning_rules.null_invalid_foreign_keys.py module
+
+The intent of this unit test is to ensure that any invalid foreign keys are nulled out while
+the remaining rows of the table are unchanged. A valid foreign key means that an existing
+foreign key already exists in the table it references. An invalid foreign key means that there
+is NOT an existing foreign key in the table it references
+"""
+
+# Python imports
+import unittest
+
+# Project imports
+import resources
+from constants import bq_utils as bq_consts
+from constants.cdr_cleaner import clean_cdr as cdr_consts
+import cdr_cleaner.cleaning_rules.null_invalid_foreign_keys as nifk
+
+
+class NullInvalidForeignKeys(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        self.project_id = 'foo_project'
+        self.dataset_id = 'bar_dataset'
+        self.sandbox_id = 'baz_sandbox'
+        self.client = None
+
+        self.rule_instance = nifk.NullInvalidForeignKeys(
+            self.project_id, self.dataset_id, self.sandbox_id)
+
+        self.assertEqual(self.rule_instance.project_id, self.project_id)
+        self.assertEqual(self.rule_instance.dataset_id, self.dataset_id)
+        self.assertEqual(self.rule_instance.sandbox_dataset_id, self.sandbox_id)
+
+    def test_setup_rule(self):
+        # Test
+        self.rule_instance.setup_rule(self.client)
+
+        # No errors are raised, nothing will happen
+
+    def test_get_query_specs(self):
+        # Pre conditions
+        self.assertEqual(self.rule_instance.affected_tables,
+                         resources.CDM_TABLES)
+
+        queries_list = []
+        sandbox_queries_list = []
+
+        for table in self.rule_instance.affected_tables:
```

Comment:
This unit test may not be sufficient because it is almost identical to `get_query_specs`.